plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -30,11 +32,12 @@ func createContainerKeyFromMetric(tags map[string]string) string {
 	return k8sutil.CreateContainerKey(namespace, podName, containerName)
 }
 
-const (
-	// kubeAllowedStringAlphaNums holds the characters allowed in replicaset names from as parent deployment
+var (
+	// deploymentAllowedRegExp holds the characters allowed in replicaset names from as parent deployment
 	// https://github.com/kubernetes/apimachinery/blob/master/pkg/util/rand/rand.go#L83
-	kubeAllowedStringAlphaNums = "bcdfghjklmnpqrstvwxz2456789"
-	cronJobAllowedString       = "0123456789"
+	deploymentAllowedRegExp = regexp.MustCompile(`^[b-hj-np-tv-xz24-9]+$`)
```

Comment:
I didn't realize until now that the series of strings came from `rand.go`.  Would it be more future proof if we utilized `rand.SafeEncodeString()` instead of maintaining our own logic that could break if apimachinery changes their implementation?

something like `suffix == rand.SafeEncodeString(suffix)`?

Reply:
I don't want to get bogged down by the design choices of k8s for their naming convention in apimachinery, which they could change at any point. As an aside, though, that code referenced hasn't been touched in any commits for over 4 years.

What is the agent's usage of the string? We've already proven that we're fragile to changes. In hindsight, I should have thought about this more to begin with, and I also didn't notice that this is pulled directly from a constant internally in k8s. If we're going to tackle this with the angle of future proofing from any updates that come to how apimachinery is creating identifiers, I think we have to take a step back and look at how we're using it. I'll have to read through the code later to see what it's used for. 

---

the `SafeEncodeString()` function is **not** idempotent. Although deterministic, and I think technically reversible, it is not idempotent, so I don't think that there is ever a scenario (at least not a reasonable scenario) where `suffix == rand.SafeEncodeString(suffix)` holds true.

https://go.dev/play/p/Jkbn-ceoaEf

Reply:
Same thought. At least for rand.SafeEncodeString I believe. For now, we can merge this PR and create an issue for the independent between packages to track it. The only thing that I can offer is to track the milestone and see any updates relating to ```naming convention``` though. If we have any dependencies with K8s then I guess we can utilize ```dependabot alert``` to keep track automatically for us though but I guess we not

Reply:
Rather than trying to guess at what k8s could change their naming convention to in the future, I think what we can do is the following:
1. Add two integration tests.
    1. One integration test that brings up an EKS cluster that uses k8s < v1.21, and verifies that the pod name is accurate
    2. One integration test that brings up an EKS cluster **on the latest version** and verifies that the pod name is accurate
2. Update the GitHub workflow to run integration tests on a weekly or bi-weekly schedule, so that we are consistently running the EKS integration tests, and not waiting for merges to the main branch to execute them

This way, we (1) provide further confidence that the change works, (2) bolster our GitHub integration test count, (3) continuously monitor integration test regression on GitHub, and (4) have a better chance of finding out about changes in k8s name changes in the future. Unless k8s provides a public method to extract the pod name themselves, then we are stuck in a situation where we have to infer the ruleset for the name generation regardless. Could be worthwhile to check if they already have an issue for that.

Reply:
Yes. @SaxyPandaBear  has offered the idea of integration test for me and I have created it in a separate PR so no worries about holding up this PR.